packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -552,3 +600,12 @@ export interface SecretStringGenerator {
    */
   readonly generateStringKey?: string;
 }
+
+/** Returns the secret name if defined, otherwise attempts to parse it from the ARN. */
+export function parseSecretName(construct: IConstruct, secretArn: string, secretName?: string) {
+  const secretNameFromNameOrArn = secretName ?? Stack.of(construct).parseArn(secretArn).resourceName;
```

Comment:
This will include the generated suffix. Is that on purpose?

Reply:
I _could_ strip out everything after the last `-` from the resourceName; I was a bit worried about people providing "incomplete" ARNs (e.g., ARN format but without the suffix) and that breaking things. But I could always enforce that the arn must have a resourceName and that resourceName must contain at least one `-`.